Adding Adds support for Terraform destroy under  command

trutled3 opened the pull request:
## Description

This pull request adds the capability to create a destroy run by setting the `data.attributes.is-destroy` key.

<br class="Apple-interchange-newline">

## Testing plan

1.  Build tfci, `go build`
2. `export TF_API_TOKEN=<TOKEN>`
3. `export TF_CLOUD_ORGANIZATION=<ORG>`
4. Run tfci, `tfci -hostname=terraform.duke-energy.com -organization=rdpautomation-test run create -workspace=airflow-sbx -is-destroy=true

## External links

- [API documentation](https://developer.hashicorp.com/terraform/enterprise/api-docs/run#request-body)
- [tfe GO package, RunCreateOptions](https://pkg.go.dev/github.com/hashicorp/go-tfe#RunCreateOptions)

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/tfc-workflows-tooling?pullRequest=80)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/tfc-workflows-tooling?pullRequest=80)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/tfc-workflows-tooling?pullRequest=80).</sub>

trutled3 commented:
The CLA has been signed, however the hashicorp-cla is not updating to reflect. Please see the attached link: https://cla.hashicorp.com/hashicorp/tfc-workflows-tooling?pullRequest=80

aaabdelgany reviewed (commented):
Overall this looks good to me. Can you please remove the binary from the PR (we'll add this to the `.gitignore` file) and bump the changelog entry to `v1.2.0` since this adds new functionality?

aaabdelgany reviewed (commented):
My apologies @trutled3, could you actually bump the changelog entry to the unreleased section and remove `v1.2.0` ?

With regards to the cli check - the cla bot may not be recognizing your signature if you are using a different email than the one used in the commits(it looks like they were authored by a github account with the username `trutled`. Can you confirm if you used the email associated with that account?

trutled3 commented:
No worries @aaabdelgany , I may have misunderstood you. I have bumped the change to the unreleased section and removed the v1.2.0 section.

I have also updated the git email to this Github account, it appears I was using a different git config when I originally created the fork.

aaabdelgany reviewed (commented):
No problem whatsoever @trutled3, I was hoping the cla job would pick up that you updated your email and complete accordingly but that hasn't happened 😞 . Can you confirm that you tried to sign it again after updating your email? If so, I'll have to dig further to see what's going on.

trutled3 commented:
Hey @aaabdelgany , it does look like it picked up my signature from this account, but since I pushed commits from my other account, it is wanting me to sign it from both. Would it be best to re-fork and submit a PR with the correct git config?

aaabdelgany commented on docs/ADOPTION.md:
```diff
@@ -56,6 +57,15 @@ With Tfci and Terraform Cloud API driven runs:
+Terraform Cloud CLI can execute a destroy run with, `terraform plan -destroy -out=destroy.tfplan` followed by `terraform apply destroy.tfplan`, that will also upload the configuration and start a new Terraform Cloud run that will plan and destroy.
+
+With Tfci and Terraform Cloud API driven runs:
+- Upload terraform configuraiton as a ConfigurationVersion
```
typo: configuration

aaabdelgany reviewed (commented):
Thank you for the follow up @trutled3 ! I've tested this locally and verified that it works. I also realized that there wasn't a unit test included as part of this PR, can you add a unit test to `internal/cloud/run_test.go`?

trutled3 commented:
Ah, great catch. Thank you @aaabdelgany . I have fixed that and added a unit test for destroy run for your review.

aaabdelgany reviewed (approved):
LGTM, thank you for the contribution @trutled3!
